python/plugins/processing/script/ScriptEditorDialog.py

```diff
@@ -194,10 +197,28 @@ def setHasChanged(self, hasChanged):
 
     def runAlgorithm(self):
         d = {}
-        exec(self.editor.text(), d)
+        try:
+            exec(self.editor.text(), d)
+        except Exception as e:
+            error = QgsError(traceback.format_exc(), "Processing")
+            QgsErrorDialog.show(error,
+                                self.tr("Execution error")
+                                )
+            return
+
+        alg = None
+        for k, v in d.items():
+            if inspect.isclass(v) and issubclass(v, QgsProcessingAlgorithm) and v.__name__ != "QgsProcessingAlgorithm":
```

Comment:
I wonder if we need to add "QgsProcessingFeatureBasedAlgorithm" to the name check here - I can imagine cases where a script algorithm could/should be based off QgsProcessingFeatureBasedAlgorithm also.

Reply:
I also thought about adding feature-based alg here. Will add it now.